mmengine/config/utils.py

```diff
@@ -144,3 +149,299 @@ def visit_Assign(self, node):
             return None
         else:
             return node
+
+
+def _is_builtin_module(module_name: str) -> bool:
+    """Check if a module is a built-in module.
+
+    Arg:
+        module_name: name of module.
+    """
+    if module_name.startswith('.'):
+        return False
+    if module_name.startswith('mmengine.config'):
+        return True
+    spec = find_spec(module_name.split('.')[0])
+    # Module not found
+    if spec is None:
+        return False
+    origin_path = getattr(spec, 'origin', None)
+    if origin_path is None:
+        return False
+    origin_path = osp.abspath(origin_path)
+    if ('site-package' in origin_path
+            or not origin_path.startswith(PYTHON_ROOT_DIR)):
+        return False
+    else:
+        return True
+
+
+class ImportTransformer(ast.NodeTransformer):
+    """Convert the import syntax to the assignment of
+    :class:`mmengine.config.LazyObject` and preload the base variable before
+    parsing the configuration file.
+
+    Since you are already looking at this part of the code, I believe you must
+    be interested in the mechanism of the ``lazy_import`` feature of
+    :class:`Config`. In this docstring, we will dive deeper into its
+    principles.
+
+    Most of OpenMMLab users maybe bothered with that:
+
+        * In most of popular IDEs, they cannot navigate to the source code in
+          configuration file
+        * In most of popular IDEs, they cannot jump to the base file in current
+          configuration file, which is much painful when the inheritance
+          relationship is complex.
+
+    In order to solve this problem, we introduce the ``lazy_import`` mode.
+
+    A very intuitive idea for solving this problem is to import the module
+    corresponding to the "type" field using the ``import`` syntax. Similarly,
+    we can also ``import`` base file.
+
+    However, this approach has a significant drawback. We need to trigger the
+    import logic to parse the configuration file, which can be time-consuming.
+    This also implies that we have to download numerous dependencies to parse
+    the configuration file. Yet, there may be instances where we simply want
+    to use string-like fields in the configuration rather than actually
+    importing them.
```

Comment:
```suggestion
    However, this approach has a significant drawback. We need to trigger the
    import logic to parse the configuration file, which can be time-consuming.
    This also implies that we have to install numerous dependencies to parse
    the configuration file. Yet, there may be instances where we simply want
    to use string-like fields in the configuration rather than actually
    importing them.
```